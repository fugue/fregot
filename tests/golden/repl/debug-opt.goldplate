{
    "notes": [
        "This actually takes more steps than `debug-noopt` since we fully",
        "calculate the indexed comprehension!  But that is what we're testing."
    ],
    "command": "fregot",
    "arguments": ["repl", "-O", "--no-history-file"],
    "asserts": [
        {"exit_code": 0},
        {"stderr": "${GOLDPLATE_NAME}.stderr"},
        {"stdout": "${GOLDPLATE_NAME}.stdout"}
    ],
    "stdin": [
        ":load ../opt/comprehension-index.rego",
        ":break exposed_ports_by_interface",
        "access_twice {exposed_ports_by_interface[\"eth0\"]; exposed_ports_by_interface[\"lo1\"]}",
        "access_twice",
        ":next",
        ":next",
        "mock_input.exposed[j].port",
        ":next",
        "mock_input.exposed[j].port",
        ":next",
        "mock_input.exposed[j].port",
        ":next",
        "mock_input.exposed[j].port",
        ":next",
        ":next",
        ":next"
    ]
}
