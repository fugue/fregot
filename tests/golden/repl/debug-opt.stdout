repl% comprehension_index% comprehension_index% comprehension_index% 32|   intf := mock_input.exposed[i].interface
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
comprehension_index(debug)% 35|     mock_input.exposed[j].interface = intf
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
comprehension_index(debug)% 36|     port := mock_input.exposed[j].port
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
comprehension_index(debug)% = 8080
| intf = "eth0"
| j = 0
comprehension_index(debug)% 36|     port := mock_input.exposed[j].port
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
comprehension_index(debug)% = 8081
| intf = "eth0"
| j = 1
comprehension_index(debug)% 36|     port := mock_input.exposed[j].port
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
comprehension_index(debug)% = 443
| intf = "eth1"
| j = 2
comprehension_index(debug)% 36|     port := mock_input.exposed[j].port
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
comprehension_index(debug)% = 5000
| intf = "lo1"
| j = 3
comprehension_index(debug)% 1| access_twice {exposed_ports_by_interface["eth0"]; exposed_ports_by_interface["lo1"]}
                                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
comprehension_index(debug)% (debug) = true
(debug) finished
comprehension_index% comprehension_index% 